packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -299,95 +290,71 @@ export function getNetworkLogo(chainId: number) {
   }
 }
 
-export type SwitchChainProps = {
-  chainId: number
-  provider: ExtendedWeb3Provider
-  onSuccess?: () => void
-  onError?: (err?: Error) => void
-  onSwitchChainNotSupported?: (attemptedChainId: number) => void
-}
-
-const isSwitchChainSupported = (provider: ExtendedWeb3Provider) => {
-  const { provider: innerProvider } = provider
-  return innerProvider.isMetaMask || innerProvider.isImToken
+export function getSupportedNetworks(chainId = 0) {
+  return isNetwork(chainId).isTestnet
+    ? [ChainId.Goerli, ChainId.ArbitrumGoerli]
+    : [ChainId.Mainnet, ChainId.ArbitrumOne, ChainId.ArbitrumNova]
 }
 
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const noop = () => {}
-
-const onSwitchChainNotSupportedDefault = (attemptedChainId: number) => {
+const onSwitchChainNotSupported = (
+  attemptedChainId: number,
+  isAttemptingToPerformTx: boolean
+) => {
   const isDeposit = isNetwork(attemptedChainId).isEthereum
   const targetTxName = isDeposit ? 'deposit' : 'withdraw'
   const networkName = getNetworkName(attemptedChainId)
 
+  const message = isAttemptingToPerformTx
+    ? `Please connect to ${networkName} on your wallet before signing your ${targetTxName} transaction.`
+    : `Please connect to ${networkName} on your wallet.`
+
   // TODO: show user a nice dialogue box instead of
   // eslint-disable-next-line no-alert
-  alert(
-    `Please connect to ${networkName} to ${targetTxName}; make sure your wallet is connected to ${networkName} when you are signing your ${targetTxName} transaction.`
-  )
+  alert(message)
 }
 
-export async function switchChain({
-  chainId,
-  provider,
-  onSuccess = noop,
-  onError = noop,
-  onSwitchChainNotSupported = onSwitchChainNotSupportedDefault
-}: SwitchChainProps) {
-  // do an early return if switching-chains is not supported by provider
-  if (!isSwitchChainSupported(provider)) {
-    onSwitchChainNotSupported?.(chainId)
+/**
+ * Function to invoke when an error is thrown while attempting to switch network.
+ * https://wagmi.sh/react/hooks/useSwitchNetwork#onerror-optional
+ * @param error
+ * @param param1 - `{ chainId: number }`
+ * @param context - default value `{ isAttemptingToPerformTx: false }`
+ */
+export function handleSwitchNetworkError(
+  error: any,
+  { chainId }: SwitchNetworkArgs,
+  context: unknown = { isAttemptingToPerformTx: false }
+) {
+  const { isAttemptingToPerformTx } = context as {
+    isAttemptingToPerformTx: boolean
+  }
+  if (isUserRejectedError(error)) {
     return
   }
+  if (error.name === 'SwitchChainNotSupportedError') {
+    onSwitchChainNotSupported(chainId, isAttemptingToPerformTx)
+  }
+  Sentry.captureException(error)
```

Comment:
Probably shouldn't capture the error in case of switch chain not supported

Reply:
moved into else clause